stl/inc/memory

```diff
@@ -1504,6 +1577,117 @@ private:
     }
 };
 
+#if _HAS_CXX20
+template <class _Ty, size_t _Size>
+void _Uninitialized_copy_md(const _Ty (&_In)[_Size], _Ty (&_Out)[_Size]) {
+    if constexpr (_STD rank_v<_Ty> > 0) {
+        for (size_t _It = 0; _It < _Size; ++_It)
+            _Uninitialized_copy_md(_In[_It], _Out[_It]);
+    } else {
+        _STD uninitialized_copy_n(_In, _Size, _Out);
+    }
+}
+
+template <class _Ty>
+void _Uninitialized_value_construct_md_n(_Ty* _Out, size_t _Size) {
+    if constexpr (_STD rank_v<_Ty> > 0) {
+        for (size_t _It = 0; _It < _Size; ++_It)
+            _Uninitialized_value_construct_md_n(_Out[_It], _STD extent_v<_Ty>);
+    } else {
+        _Uninitialized_value_construct_n_unchecked1(_Out, _Size);
+    }
+}
+
+template <class _Ty>
+void _Reverse_destroy_md_n(_Ty* _Arr, size_t _Size) {
+    size_t _It = _Size;
+    do {
+        --_It;
+        if constexpr (_STD rank_v<_Ty> > 0) {
+            _Reverse_destroy_md_n(_Arr[_It], _STD extent_v<_Ty>);
+        } else {
+            _Destroy_in_place(_Arr[_It]);
+        }
+    } while (_It > 0);
+}
+
+// CLASS TEMPLATE _Ref_count_unbounded_array
+template <class _Ty>
+class _Ref_count_unbounded_array : public _Ref_count_base { // handle reference counting for unbounded array
+                                                            // in control block, no allocator
+public:
+    explicit _Ref_count_unbounded_array(size_t _Count) : _Ref_count_base(), _Size(_Count) {
+        _Uninitialized_value_construct_md_n(_Get_ptr(), _Size);
+    }
+
+    explicit _Ref_count_unbounded_array(size_t _Count, const _STD remove_extent_t<_Ty>& _Val)
+        : _Ref_count_base(), _Size(_Count) {
+        if constexpr (_STD rank_v<_Ty> > 1) {
+            for (size_t _It = 0; _It < _Size; ++_It)
+                _Uninitialized_copy_md(_Val, _Get_ptr()[_It]);
+        } else {
+            _STD uninitialized_fill_n(_Get_ptr(), _Size, _Val);
+        }
+    }
+
+    ~_Ref_count_unbounded_array() {
+        // nothing to do, _Storage was already destroyed in _Destroy

```

Comment:
If there's nothing to do we shouldn't be inhibiting trivial destruction.

Reply:
Fixed.